source/plugin/data/serialization.rst

```diff
@@ -123,3 +125,51 @@ from persistent files using the :doc:`Configurate Library <../configuration/inde
         final DataView container = translator.translate(node);
         return deserializeHealth(container);
     }
+
+
+DataFormat
+==========
+An alternative to DataTranslators is to use :javadoc:DataFormat, which allows you to store a ``DataContainer`` in HOCON, JSON or NBT format. You can also recreate DataConainters using ``DataFormats``.
+
+For example, you could use ``DataFormat`` to create a json repusentation of a ``DataContainer``, which you could then easily store in a database as JSON. ``DataFormat`` can then be used to recreate the original ``DataContainer`` from json where required.
+
+For this example we will use :javadoc:`DataFormats#JSON` to translate a ItemStackSnapshot to JSON and back! 
+
+**Imports for code examples**
+    import org.spongepowered.api.Sponge;
+    import org.spongepowered.api.data.DataContainer;
+    import org.spongepowered.api.data.DataQuery;
+    import org.spongepowered.api.data.DataView;
+    import org.spongepowered.api.data.persistence.DataBuilder;
+    import org.spongepowered.api.data.persistence.DataFormat;
+    import org.spongepowered.api.data.persistence.DataFormats;
+    import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+
+    import java.io.ByteArrayOutputStream;
+    import java.io.IOException;
+    import java.util.Optional;
+
+**Code Example: Serializing a ItemStackSnapshot to JSON format**
+    
+.. code-block:: java
+
+    public ByteArrayOutputStream toJsonFormat(ItemStackSnapshot itemStack){
```

Comment:
Can you add a space before `{`

Reply:
I also think it would be easier to understand if you don't return the `ByteArrayOutputStream`, but rather use a specific example like writing a `ItemStackSnapshot` to a file. I don't think anyone would use the method like this.

Reply:
done in the new pr